sdk/typescript/src/zklogin/publickey.ts

```diff
@@ -75,5 +87,82 @@ export function toZkLoginPublicIdentifier(
 	tmp.set([issBytes.length], 0);
 	tmp.set(issBytes, 1);
 	tmp.set(addressSeedBytesBigEndian, 1 + issBytes.length);
-	return new ZkLoginPublicIdentifier(tmp);
+	return new ZkLoginPublicIdentifier(tmp, options);
+}
+
+const VerifyZkLoginSignatureQuery = graphql(`
+	query Zklogin(
+		$bytes: Base64!
+		$signature: Base64!
+		$intent_scope: ZkLoginIntentScope!
+		$author: SuiAddress!
+	) {
+		verifyZkloginSignature(
+			bytes: $bytes
+			signature: $signature
+			intentScope: $intent_scope
+			author: $author
+		) {
+			success
+			errors
+		}
+	}
+`);
+
+async function graphqlVerifyZkLoginSignature({
+	address,
+	bytes,
+	signature,
+	client = new SuiGraphQLClient({
+		url: 'https://sui-testnet.mystenlabs.com/graphql',
+	}),
+}: {
+	address: string;
+	bytes: string;
+	signature: string;
+	client?: SuiGraphQLClient;
+}) {
+	const resp = await client.query({
+		query: VerifyZkLoginSignatureQuery,
+		variables: {
+			bytes,
+			signature,
+			intent_scope: 'PERSONAL_MESSAGE',
```

Comment:
Should have noticed this earlier, but this won't actually work as is.

The public keys verifyPersonalMessage method will call verifyWithIntent, which then calls verify with the digest of the intent message.  The `verify` method we implemented on zksend public key is intended for both personal messages and transactions, and is passed only the signature, and the digest of the intent message,  rather than the raw message bytes.

Reply:
graphql always accepts the raw personal message bytes not the digest.

Reply:
The issue is that PublicKeyInitData.verify doesn't get the message bytes, it gets the digest, so while `graphqlVerifyZkLoginSignature` will work, `pk.verifyPersonalMessage` will not